src/target/spirv/spirv_support.h

```diff
@@ -0,0 +1,220 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+ * \file spirv_support
+ *
+ * \brief Utility for determining which spirv capabilities a TVM
+ * target supports.
+ */
+#ifndef TVM_TARGET_SPIRV_SPIRV_SUPPORT_H_
+#define TVM_TARGET_SPIRV_SPIRV_SUPPORT_H_
+
+#include <tvm/target/target.h>
+
+namespace tvm {
+namespace codegen {
+
+/*! \brief Represents which support a Vulkan driver has that are relevant to codegen */
+struct SPIRVSupport {
+  /*! \brief Determine spirv capabilities from a vulkan target.
+   */
+  explicit SPIRVSupport(Target target);
+
+  /*!
+   * \brief The supported operations
+   *
+   * Vulkan extension: VK_KHR_driver_properties
+   * Minimum vulkan version: 1.1
+   *
+   * The supportedOperations bitflags from
+   * VkPhysicalDeviceSubgroupProperties.
+   *
+   * Requires vulkan 1.1 or higher to use.  If the
+   * VK_KHR_driver_properties extension is not present in order to
+   * query this value, or if the driver does not support vulkan 1.0,
+   * then this value will be set to 0.
+   *
+   */
+  uint32_t supportedSubgroupOperations{0};
```

Comment:
note: stl_style to be consistent with the rest of the codebase

Reply:
Thank you, and I will change the names.  I had been attempting to have the names here mimic the names of the spirv/vulkan features that they enable, but I can see how that would be inconsistent elsewhere.

Reply:
Member variable names have been updated, along with adding additional documentation on where they should be derived from on the vulkan side.